vortex-buffer/src/lib.rs

```diff
@@ -135,13 +149,18 @@ impl AsRef<[u8]> for Buffer {
     }
 }
 
-impl From<&[u8]> for Buffer {
-    fn from(value: &[u8]) -> Self {
-        // We prefer Arrow since it retains mutability
+impl From<&'static [u8]> for Buffer {
```

Comment:
Why remove the support for `&[u8]` and `&str`?

Reply:
The implicit copy is a bit gross, prefer to be explicit. And for statics, we don't do a copy. So it made it too easy to accidentally copy static refs.

Reply:
seems like arrow still copies 'static things into the buffer